iac-extensions/arm/src/test/resources/checks/HardcodeApiVersionCheck/resources.json

```diff
@@ -0,0 +1,31 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "parameters": {
+    "customApiVersion": {
+      "type": "string"
+    }
+  },
+  "variables": {
+    "customApiVersion": "[first(providers(‘Microsoft.Compute’,’virtualMachines’).apiVersions)]"
+  },
+  "resources": [
+    {
+      "apiVersion": "[parameters('customApiVersion')]",
+      "type": "Microsoft.Compute/virtualMachines",
+      "name": "nonCompliantResource1",
+      "location": "[resourceGroup().location]"
+    },
+    {
+      "apiVersion": "[variables('customApiVersion')]",
+      "type": "Microsoft.Compute/virtualMachines",
+      "name": "nonCompliantResource2",
+      "location": "[resourceGroup().location]"
+    },
+    {
+      "apiVersion": "2023-09-01",
```

Comment:
Maybe we should add additional test cases to document the behavior?
What if we encounter `[concat('2023','-09-01')]`. ( We should raise, but I believe it's interesting to document in test)